### PR TITLE
Sth pythonmods

### PR DESCRIPTION
**Pull request by sth0**
Updated duplicity to version 0.8.17 and use of Python 3.8

Updated these packages include Python 3.8 and corrected some source URLs to more readable form
alabaster-py
astroid-py
babel-py
bcrypt-py
boto-py
duplicity
fasteners-py
isort-py
lazy-object-proxy-py
logilab-common-py
mccabe-py
monotonic-py
oauthlib-py
paramiko-py
pexpect-py
ptyprocess-py
pycrypto-py
pylint-py
pytest-runner-py
requests-oauthlib-py
snowballstemmer-py
sphinx-py
sphinxcontrib-websupport-py
tox-py
virtualenv-py

Updated astroid-py to upstream 2.2.5
Updated boto-py to upstream version 2.49.0
Updated sphinx-py to upstream version 3.1.0
Updated virtualenv-py to upstream 16.7.5

### CODE REVIEW

**Comment by sth0** [comment]
Duplicity passed the self test and builds in maintainer mode and works on OS 10.11 and 10.14

**Comment by nieder** [comment]
* boto-py adds support for py33 and py34, but py33 stopped being supported at OS X 10.8. py34 is EOL upstream as well (though still in Fink for 10.9-present). If you're adding new python support, I'd just do 37 and 38 unless something else actually needs py34-py36. Adding py34-
* For the shebangs in duplicity.patch, you don't need `#!/usr/bin/env @PREFIX@/bin/python3.8`. These can be simplified to just `#!@PREFIX@/bin/python3.8`

* updating tox-py and sphinx-py are a mess (not here specifically, but in general). It would be far better to split those two packages into separate issues since their upgrade is much more involved than just adding a new variant. See #258 for example. @dhomeier

**Comment by sth0** [comment]
boto-py duly noted and will fix.
duplicity - was on the principle of do least changes, upstream has the /usr/bin/env and I was just adding the PREFIX and python version to make it work.  I see that /bin/env is called to make handling of arguments to the python command parse better but since there are no arguments to python it does not matter.  I will editor out the calls to /bin/env.
sphinx and tox will be a separate branch then.  Again I was trying to do least changes.

**Comment by sth0** [comment]
pushed corrections and created 2 branches for tox and sphinx

**Comment by sth0** [comment]
Until I get Tox and Sphinx working duplicity will fail its tests, but it can build and install AFAIK.  Can the pull request be done with the current update that removes Tox and Sphinx though knowing the duplicity tests will fail?  I reviewed the issues with Tox and I am not sure I follow the issue, possibly some of the issues have been fixed with the newer release.  I will go back to that thread and see if I can revive it.

**Comment by sth0** [comment]
Can I get these low level patches reviewed?  I have the actual package that I am working on (duplicity) updated significantly for Python 3.8 but until all the utilities are updated I am blocked.

**Review by dmacks — COMMENTED** [review]
By eye it all looks good. I see two oddities that are persisting from before, may as well fix them?

File: 10.9-libcxx/stable/main/finkinfo/utils/duplicity.info [hunk 1]
```diff
@@ -13,56 +13,60 @@ BuildDepends: <<
...
     	#!/bin/sh -ev
-    	ulimit -n 5120
-		chmod 0700 `pwd`/testing/gnupg
-        %p/bin/python2.7 setup.py test
+	OSXVersion="$(sw_vers -productVersion | cut -f 2- -d .)"
+	if [[ "${OSXVersion}" < "14" ]]; then
+	    ulimit -n 8192
+	fi
+	chmod 0700 `pwd`/testing/gnupg
+        %p/bin/python3.8 setup.py test
```
[the remark this answers is not in the document]
Comment (sth0) [reply 10.9-libcxx/stable/main/finkinfo/utils/duplicity.info:63]: Thanks.   
Spacing: I think one line had tabs and one had spaces.  Now both are tabs.

I will remove the superfluous call to pwd as you suggest.  BUT that test for OS version that I wrote a year ago will now fail on Mac OS 11.0.  Is there a better way to test for operating system versions?

[the remark this answers is not in the document]
Comment (sth0) [reply 10.9-libcxx/stable/main/finkinfo/utils/duplicity.info:63]: Figured it out:     if dpkg --compare-versions `sw_vers -productVersion` \< 10.14 ; then
but testing can't happen unless there is a tox-py38 and sphinx-py38

[the remark this answers is not in the document]
Comment (sth0) [reply 10.9-libcxx/stable/main/finkinfo/utils/duplicity.info]: Pushed edits to duplicity.info

**Comment by nieder** [comment]
checking non-sphinx/tox/duplicity packages. Will cherry pick patch and commit directly if no issues.

**Comment by nieder** [comment]
I think I have everything except tox/sphinx/duplicity from here. astroid-py27 couldn't go all the way up to 2.2.5, so I made 1.6.6 which is the last one for it. And there were some test-related py38 holes that needed to be filled (flake8, pytest-relaxed, and a couple others).

**Comment by nieder** [comment]
OK, I've pushed all your non-tox/sphinx/duplicity updates with minor tweaks to fix current master.
Suggestion:
1) close this PR,
2) make 2 separate PRs for tox and sphinx,
2a) for sphinx, move as many as possible of the sphinxcontrib-* modules into their own separate packages, like was done with sphinxcontrib-websupport-py*. sphinx_rtd_theme should still be part of sphinx-py itself because it's needed for testing.
3) make a separate PR for duplicity.

This will help tracking changes and understanding what's being updated where.

**Comment by sth0** [comment]
Got it.  Good catch with astroid.  I thought I had tested all the 2.7 stuff on my old machine but that slipped by.
I am working on tox and sphinx and started separate branches.
I did not mean for duplicity to be part of this PR but obviously snuck in during testing.  I plan to submit that last as a final goal.
Will close this.  Thanks.

**Comment by nieder** [comment]
For tox, see #258 for some discussion on updating our current 3.0.0 to (now old 3.2.1). Don't know how much of that is still needed for upstream's 3.20.1, but some might.

**Pull request closed without merging**